pkg/systemd/service-details.jsx

```diff
@@ -0,0 +1,506 @@
+ * Optional props:
+ *  - confirmText
+ *     Text of the button for confirming the action
+ *  - confirmAction
+ *     Action to be executed when the action is confirmed
+ */
+class ServiceConfirmDialog extends React.Component {
+    render() {
+        return (
+            <Modal show>
+                <Modal.Header>
+                    <Modal.Title>{this.props.title}</Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                    {this.props.message}
+                </Modal.Body>
+                <Modal.Footer>
+                    <Button bsStyle='default' className='btn-cancel' onClick={this.props.close}>
+                        { _("Cancel") }
+                    </Button>
+                    { this.props.confirmText && this.props.confirmAction &&
+                        <Button bsStyle='danger' onClick={this.props.confirmAction}>
+                            {this.props.confirmText}
+                        </Button>
+                    }
+                </Modal.Footer>
+            </Modal>
+        );
+    }
+}
+ServiceConfirmDialog.propTypes = {
+    title: PropTypes.string.isRequired,
+    message: PropTypes.string.isRequired,
+    close: PropTypes.func.isRequired,
+    confirmText: PropTypes.string,
+    confirmAction: PropTypes.func,
+};
+
+/*
+ * React template for showing possible service action (in a kebab menu)
+ * Required props:
+ *  - masked
+ *     Unit is masked
+ *  - active
+ *     Unit is active (running)
+ *  - enabled
+ *     Unit is enabled
+ *  - isStatic
+ *     Unit is static
+ *  - actionCallback
+ *      Method for calling unit methods like `UnitStart`
+ *  - fileActionCallback
+ *      Method for calling unit file methods like `EnableUnitFiles`
+ *  - disbaled
+ *      Button is disabled
+ */
+class ServiceActions extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            dialogMaskedOpened: false,
+        };
+    }
+
+    render() {
+        let actions = [];
+
+        // If masked, only show unmasking and nothing else
+        if (this.props.masked)
+            actions.push(
+                <MenuItem key="unmask" onClick={() => this.props.fileActionCallback("UnmaskUnitFiles", undefined)}>{ _("Allow running (unmask)") }</MenuItem>
+            );
+        else { // All cases when not masked
+            // Only show stop when running but not enabled
+            if (this.props.active && !this.props.enabled) {
+                actions.push(
+                    <MenuItem key="stop" onClick={() => this.props.actionCallback("StopUnit")}>{ _("Stop") }</MenuItem>,
+                );
+            }
+            if (!this.props.isStatic && (this.props.enabled || this.props.active)) {
+                if (actions.length > 0)
+                    actions.push(
+                        <MenuItem key="divider1" divider />
+                    );
+                if (this.props.active)
```

Comment:
nitpick; We generally tend to put the if else bodys inside curly braces when they are multiline.